tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs

```diff
@@ -434,12 +434,23 @@ public void Double_negative_read()
     [InlineData(typeof(int?), 0)]
     [InlineData(typeof(int?), -15)]
     [InlineData(typeof(int?), 42)]
+    [InlineData(typeof(TestEnum), TestEnum.EnumValue0)]
+    [InlineData(typeof(TestEnum), TestEnum.EnumValue1)]
+    [InlineData(typeof(TestEnum?), null)]
+    [InlineData(typeof(TestEnum?), TestEnum.EnumValue0)]
+    [InlineData(typeof(TestEnum?), TestEnum.EnumValue1)]
     public void ClrTypeMappingTest(Type valueType, object value)
     {
         var methodInfo = this.GetType().GetMethod(nameof(ClrTypeMappingTestImpl), BindingFlags.Instance | BindingFlags.NonPublic);
         methodInfo.MakeGenericMethod(valueType).Invoke(this, new[] { value });
     }
 
+    private enum TestEnum
+    {
+        EnumValue0 = 0,
+        EnumValue1 = 1
+    }
+
```

Comment:
Can we include some test coverage for inserts of entities with Enums? 

Right now the test uses the Mongo C# Driver to insert the data for the test which is fine for exercising the query and materialization paths but insert/update path should also have coverage.

Reply:
Done.